tests/unit/labels/LabelsEncoder_unittest.cc

```diff
@@ -0,0 +1,170 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/labels/BinaryLabelEncoder.h>
+#include <shogun/labels/LabelEncoder.h>
+#include <shogun/labels/MulticlassLabels.h>
+#include <shogun/labels/MulticlassLabelsEncoder.h>
+using namespace shogun;
+
+TEST(BinaryLabelEncoder, fit_transform)
+{
+	auto label_encoder = std::make_shared<BinaryLabelEncoder>();
+	SGVector<int32_t> vec{-1, -1, 1, -1, 1};
+	auto origin_labels = std::make_shared<BinaryLabels>(vec);
+	auto unique_vec = label_encoder->fit(origin_labels);
+	EXPECT_EQ(-1, unique_vec[0]);
+	EXPECT_EQ(1, unique_vec[1]);
+
+	auto result_labels = label_encoder->transform(origin_labels);
+	auto result_vec = result_labels->as<BinaryLabels>()->get_labels();
+
+	SGVector<int32_t> expected_res{-1, -1, 1, -1, 1};
+	for (int i = 0; i < 5; i++)
+	{
+		EXPECT_EQ(expected_res[i], result_vec[i]);
+	}
+
+	auto inv_result = label_encoder->inverse_transform(result_labels)
+	                      ->as<BinaryLabels>()
+	                      ->get_labels();
+
+	for (int i = 0; i < 5; i++)
+	{
+		EXPECT_EQ(vec[i], inv_result[i]);
+	}
+}
+
+TEST(BinaryLabelEncoder, labels_not_neg1_or_1)
+{
+	auto label_encoder = std::make_shared<BinaryLabelEncoder>();
+	SGVector<int32_t> vec{-100, 200, -100, 200, -100};
+	auto origin_labels = std::make_shared<BinaryLabels>(vec);
+	auto unique_vec = label_encoder->fit(origin_labels);
+	EXPECT_EQ(-100, unique_vec[0]);
+	EXPECT_EQ(200, unique_vec[1]);
+
+	auto result_labels = label_encoder->transform(origin_labels);
+	auto result_vec = result_labels->as<BinaryLabels>()->get_labels();
+	SGVector<int32_t> expected_vec{-1, 1, -1, 1, -1};
+	for (int i = 0; i < 5; i++)
+	{
+		EXPECT_EQ(expected_vec[i], result_vec[i]);
+	}
+
+	auto inv_result = label_encoder->inverse_transform(result_labels)
+	                      ->as<DenseLabels>()
+	                      ->get_labels();
+
+	for (int i = 0; i < 5; i++)
+	{
+		EXPECT_EQ(vec[i], inv_result[i]);
+	}
+
+	SGVector<int32_t> test_vec{-1, -1, -1, -1, -1, 1};
+	auto test_labels = std::make_shared<BinaryLabels>(test_vec);
+	auto inv_test = label_encoder->inverse_transform(test_labels)
+	                    ->as<BinaryLabels>()
+	                    ->get_labels();
+	SGVector<int32_t> expected_inv_test{-100, -100, -100, -100, -100, 200};
+	for (int i = 0; i < 6; i++)
+	{
+		EXPECT_EQ(expected_inv_test[i], inv_test[i]);
+	}
+}
+
+TEST(BinaryLabelEncoder, more_than_two_labels)
+{
+	auto label_encoder = std::make_shared<BinaryLabelEncoder>();
+	SGVector<int32_t> vec{-100, 200, -100, 200, -100, 42};
+	auto origin_labels = std::make_shared<BinaryLabels>(vec);
```

Comment:
@karlnapf doesn't BinaryLabels throw in this situation? And if not shouldn't it?
Also shouldn't this test be done with MulticlassLabels?

Reply:
I think it doesnt atm, but I am not sure.